File: ql/src/experimental/CWE-918/SSRFGood.go
package main

import (
	"net/http"
	"github.com/go-playground/validator"
)

func handler(w http.ResponseWriter, req *http.Request) {
	validate := validator.New()
	target := req.FormValue("target")
	if validate.Var(target, "alphanum")
	// GOOD: `target` is alphanumeric
	resp, err := http.Get("https://example.com/current_api/" + target)
	if err != nil {
		// error handling
	}

	// process request response
	use(resp)
}

File: javascript/ql/src/experimental/Security/CWE-918/SSRFGood.js
const axios = require('axios');
const validator = require('validator');

export const handler = async (req, res, next) => {
  const { target } = req.body;

  if (!validator.isAlphanumeric(target)) {
    return next(new Error('Bad request'));
  }

  try {
    // `target` is validated
    const response = await axios.get('https://example.com/current_api/' + target);

    // process request response
    use(response);
  } catch (err) {
    // process error
  }
};
